docs/dynamicproxy-default-parameter-values.md

```diff
@@ -0,0 +1,37 @@
+
+* **Optional parameters of any nullable type `Nullable<T>`.** Reflection will likely report (via `ParameterInfo.[Raw]DefaultValue`) a default value of `null` for such parameters, regardless of whether the actual default value in the proxied method was `null` or not.
+
+   Therefore, you cannot trust the reported default value, *unless* you know you're running on a version of Mono where the underlying issues have been resolved, or when you've double-checked the default value in the original method of the proxied type.
+
+   The underlying causes have been documented in [mono/mono#8504](https://github.com/mono/mono/issues/8504) and [mono/mono#8597](https://github.com/mono/mono/issues/8597).
+
+
+## When your code runs on .NET or .NET Core
+
+.NET (up to and including at least version 4.7.1) and .NET Core (up to and including at least version 2.1) are affected by several bugs or limitations regarding default parameter values. DynamicProxy may not be able to correctly reproduce default parameter values in the proxy type for...
+
+* **Optional parameters of any nullable type `Nullable<T>`.** On .NET 3.5 only, reflection will likely report (via `ParameterInfo.[Raw]DefaultValue`) a default value of `Missing.Value` for such parameters.
+
+   There is no easy way to quickly guess what the correct default value might be. Consider upgrading to .NET 4 or later, or double-check the default value in the original method of the proxied type.
+
+* **Optional parameters of some `struct` type `SomeStruct` having a default value of `default(SomeStruct)`.** If reflection reports (via `ParameterInfo.[Raw]DefaultValue`) a default value of `Missing.Value` for such parameters, you may safely assume that the *correct* default value is `default(SomeStruct)`.
+
+   Note that if reflection reports a default value of `null` in such cases, this is not an error, but normal `System.Reflection` behavior that is to be expected. In this case, you may also safely assume `default(SomeStruct)` to be the correct default value.
+
+   For .NET Core, the underlying cause of this problem has been documented in [dotnet/corefx#26164](https://github.com/dotnet/corefx/issues/26164). A bugfix has been merged, so this issue may disappear sometime soon after .NET Core 2.1.
+
+* **Optional parameters of some nullable `enum` type `SomeEnum?` having a non-`null` default value.** If reflection reports (via `ParameterInfo.[Raw]DefaultValue`) a default value of `Missing.Value` for such parameters, the only thing you may safely assume is that the actual default value is not `null`.
+
+   You can only find out the correct default value by double-checking the default value in the original method of the proxied type.
+
+   For .NET Core, the underlying cause of this problem has been documented in [dotnet/coreclr#17893](https://github.com/dotnet/coreclr/issues/17893).
```

Comment:
Add another bullet point for the (closed generic) enum case mentioned in the other review comment.

Reply:
Done. 🏁